### PR TITLE
update general/conflux-basics/gas.md, regarding the issue #451

### PR DESCRIPTION
**Pull request by jackleeio**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://doc.confluxnetwork.org/docs/general/CONTRIBUTING#create-a-pull-request).
- [x] **This is NOT a PR to submit translation**

- - -
This change is [Reviewable](https://reviewable.io/reviews/Conflux-Chain/conflux-documentation/453)

### CODE REVIEW

**Review by darwintree — CHANGES_REQUESTED** [review]
Thank you for your pull request. There are some changes needed before this pull request could get merged.

1. The position where the content is inserted. Currently the new part is inserted into between `How gasFee is calculated` and the subsection `gasUsed`. It is supposed to move somewhere else to retain context consistency
    image
2. the second point and third point are referring the same thing ---- sponsorship mechanism. And I don't think this is the suitable contents here. From the user's perspective, he has hardly choice making use of the sponsorship mechanism or not. And from developers' perspective, the developer did not pay less.
3. The header "How do I pay less gas?" should be changed. For example, "How can users pay less gas fee?". And relating contents can be illustrated from user's or developer's perspective, for example, optimizing contract logic, using multicall, etc

**Comment by jackleeio** [comment]
Made some changes as per your tips

**Comment by darwintree** [comment]
> Made some changes as per your tips

The modified content is good! And for the current content, I think it would be better to be listed as an individual article because the pages in `general/conflux-basics` folder will mainly focus on general concepts rather than detailed development guide. You can move the gas optimization suggestion to create a new page in general/build/contracts as you suggested in #454. Then you can link the new page in this page.

Another question is the *a series of tutorials* for gas optimization. I guess you are going to create detailed tutorials, for example, illustrate the mechanism and giving an example on how to use libraries. In that case that would be awesome. But it is also a choice that you only contain the most important key points in a single article. But anyway it depends on you.

**Comment by darwintree** [comment]
I think after the above minor change, the current content is ready to be merged. Would you like to be merged immediately or after all the relating gas optimization tutorials are ready?

File: docs/general/build/smart-contracts/introduction-to-smart-contracts.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 sidebar_position: 1
 title: Smart Contracts Development
-displayed_sidebar: generalSidebar
+# displayed_sidebar: generalSidebar
```
[the remark this answers is not in the document]
Comment (jackleeio) [reply docs/general/build/smart-contracts/introduction-to-smart-contracts.md:4]: Got it

**Review by darwintree — APPROVED** [review]
lgtm

**Comment by jackleeio** [comment]
thanks, bro

**Pull request merged**